rlclientlib/logger/async_batcher.h

```diff
@@ -151,11 +156,28 @@ namespace reinforcement_learning { namespace logger {
       }
     }
 
-    RETURN_IF_FAIL(collection_serializer.finalize(status));
-
+    if (_events_counter_status == events_counter_status::ENABLE) {
+      uint64_t buffer_start_event_number = _buffer_end_event_number;
+      _buffer_end_event_number = evt.get_event_index();
+      uint64_t original_count = (_buffer_end_event_number - buffer_start_event_number);
+      RETURN_IF_FAIL(collection_serializer.finalize(status, original_count));
+    }
+    else {
+      RETURN_IF_FAIL(collection_serializer.finalize(status));
+    }
     return error_code::success;
   }
 
+  template<typename TEvent, template<typename> class TSerializer>
+  void  async_batcher<TEvent, TSerializer>::update_event_with_index(TEvent evt,TEvent* item)  {
```

Comment:
why do you need to have second argument?
what is wrong with just
update_event_index(TEvent&& evt) {
...
     evt.set_event_index(_evbent_index)
}

Reply:
Yes, after using TEvent&& evt instead of TEvent second argument is no longer required.